stl/inc/chrono

```diff
@@ -2209,6 +2221,2949 @@ namespace chrono {
+            }
+
+            if (_Iso_year || _Two_dig_iso_year || _Iso_week) {
+                // Need to have %G or %C+%g. The century can be parsed explicitly, or derived implicitly from %y.
+                const bool _Has_complete_year{_Iso_year || ((_Century || _Two_dig_year) && _Two_dig_iso_year)};
+                if (!_Has_complete_year || !_Iso_week || !_Weekday || _Out_of_range(_Iso_week, 1, 53)) {
+                    return false;
+                }
+            }
+
+            if (_Week_u || _Week_w) {
+                // Need a weekday and year to be complete.
+                if (!_Weekday || !_Two_dig_year) {
+                    return false;
+                }
+
+                if (_Out_of_range(_Week_u, 0, 53) || _Out_of_range(_Week_w, 0, 53)) {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        template <class _Duration1, class _Duration2>
+        _NODISCARD static constexpr bool _Can_represent() {
+            using _Rep1    = typename _Duration1::rep;
+            using _Period1 = typename _Duration1::period;
+            using _Period2 = typename _Duration2::period;
+
+            // Returns whether _Duration1 can represent _Duration2{1}. Assumes 1 <= _Period2 <=
+            // 86,400, i.e., we're interested in time periods between seconds and days.
+            if constexpr (is_integral_v<_Rep1>) {
+                // Must have _Period1 <= _Period2 and numeric_limits<_Rep1>::max() >= _Period2 / _Period1. For example,
+                // std::days can't represent std::seconds, and duration<int, atto>::max() is ~9.2 seconds.
+                constexpr auto _Max_tick = static_cast<intmax_t>((numeric_limits<make_signed_t<_Rep1>>::max)());
+
+                // clang-format off
+                return ratio_less_equal_v<_Period1, _Period2>
+                    && ratio_greater_equal_v<ratio<_Max_tick, _Period2::num>, ratio_divide<ratio<1>, _Period1>>;
+                // clang-format on
+            } else if (is_floating_point_v<_Rep1>) {
+                // With the smallest possible _Period1, ratio<1,INTMAX_MAX>, one day has a tick count of
+                // 86,400*INTMAX_MAX ~= 7.97e23. This is representable by float and double, so they can always represent
+                // at least one day. On the other hand, one second with the largest possible _Period1 needs a tick count
+                // of 1/(INTMAX_MAX) ~= 1.08e-19, which is also representable in both float and double. So, both
+                // floating point types can represent durations between one second and one day, regardless of _Period1.
+                return true;
+            } else {
+                // TRANSITION: user-defined arithmetic-like types
+                return true;
+            }
+        }
+
+        template <class _DurationType>
+        _NODISCARD bool _Apply_duration_fields(_DurationType& _Result, const bool _For_time_point) {

```

Comment:
I have been looking a lot for what the second argument stands for.

Also I do not see any occurence where this is a runtime variable.

Should that actually be a template parameter and we could make it an enum with a descriptive name

Reply:
:+1: This could indeed be transformed into a template parameter, and an enum class.

The `true` call is within `_Make_time_point`, which is called by `from_stream` for `utc_time`, `sys_time`, `tai_time`, etc., so it literally means "we're parsing a `time_point`".